### PR TITLE
remove std::iterator from wrapped_compact_theta_sketch_alloc

### PR DESCRIPTION
**Pull request by jmalkin**

### CODE REVIEW

**Comment by jmalkin** [comment]
Should fix #370

**Comment by AlexanderSaydakov** [comment]
> const uint64_t& operator*() const;
> const uint64_t* operator->() const;

these should return "reference" and "pointer"

File: theta/include/theta_sketch.hpp [hunk 1]
```diff
@@ -471,8 +471,14 @@ class wrapped_compact_theta_sketch_alloc : public base_theta_sketch_alloc<Alloca
 };
 
 template<typename Allocator>
-class wrapped_compact_theta_sketch_alloc<Allocator>::const_iterator: public std::iterator<std::input_iterator_tag, uint64_t> {
+class wrapped_compact_theta_sketch_alloc<Allocator>::const_iterator {
 public:
+  using iterator_category = std::input_iterator_tag;
+  using value_type = uint64_t;
```
Comment (AlexanderSaydakov) [inline theta/include/theta_sketch.hpp:477]: const

File: theta/include/theta_sketch.hpp [hunk 2]
```diff
@@ -471,8 +471,14 @@ class wrapped_compact_theta_sketch_alloc : public base_theta_sketch_alloc<Alloca
...
+  using difference_type = void;
+  using pointer = uint64_t*;
```
Comment (AlexanderSaydakov) [inline theta/include/theta_sketch.hpp:479]: const

File: theta/include/theta_sketch.hpp [hunk 3]
```diff
@@ -471,8 +471,14 @@ class wrapped_compact_theta_sketch_alloc : public base_theta_sketch_alloc<Alloca
...
+  using reference = uint64_t;
```
Comment (AlexanderSaydakov) [inline theta/include/theta_sketch.hpp:480]: const

**Comment by AlexanderSaydakov** [comment]
While we are at this, let's add operator->() to theta_update_sketch_base theta_iterator and theta_const_iterator

File: theta/include/theta_sketch.hpp (hunk 3 above)
Comment (jmalkin) [reply theta/include/theta_sketch.hpp:480]: gcc complains (warning) about `const` here in the return type.

Comment (AlexanderSaydakov) [reply theta/include/theta_sketch.hpp:480]: right, let't not use const when returning by value

File: theta/include/theta_union_base_impl.hpp [hunk 4]
```diff
@@ -42,7 +42,7 @@ void theta_union_base<EN, EK, P, S, CS, A>::update(SS&& sketch) {
   if (sketch.get_seed_hash() != compute_seed_hash(table_.seed_)) throw std::invalid_argument("seed hash mismatch");
   table_.is_empty_ = false;
   union_theta_ = std::min(union_theta_, sketch.get_theta64());
-  for (auto& entry: sketch) {
+  for (auto entry: sketch) {
```
Comment (AlexanderSaydakov) [inline theta/include/theta_union_base_impl.hpp:45]: I think we need auto&& here too

**Review by AlexanderSaydakov — APPROVED** [review]

**Pull request merged**